Sources/OktaAnalytics/OktaAnalytics.swift

```diff
@@ -102,12 +109,199 @@ public final class OktaAnalytics: NSObject {
      - eventName: `event name` provided by client
      - withProperties: `properties/metadata` associated with event
      */
-    public static func trackEvent(_ eventName: String, withProperties: [String: String]?) {
+    public static func trackEvent(_ eventName: String, withProperties: Properties) {
         lock.readLock()
         defer { lock.unlock() }
         providers.forEach { $1.trackEvent(eventName, withProperties: withProperties) }
     }
 
+    /**
+        Starts an event scenario with the specified name.
+
+        - Parameters:
+           - eventName: The name of the event scenario to start.
+           - propertySubject: A closure that takes a `PassthroughSubject` of `Property` objects as a parameter.
+        */
+    public static func startScenario(_ scenarioName: Name, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void) -> ScenarioID {
```

Comment:
can we structure the  code which leads to simpler name, example  OktaAnalytics.scenario.start(_ Name: Name, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void ?